File: src/main/java/org/example/kalkulationsprogramm/domain/ArbeitsgangStundensatz.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@Entity
public class ArbeitsgangStundensatz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "arbeitsgang_id")
    private Arbeitsgang arbeitsgang;

    private int jahr;

    @Column(precision = 10, scale = 2, nullable = false)
    private BigDecimal satz;
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/ArbeitsgangStundensatz.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
open class ArbeitsgangStundensatz {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "arbeitsgang_id")
    open var arbeitsgang: Arbeitsgang? = null

    open var jahr: Int = 0

    @Column(precision = 10, scale = 2, nullable = false)
    open var satz: BigDecimal? = null

}